### ISSUE TITLE
Please help confirm or correct my understanding!

### ISSUE THREAD

**Issue by julianzero** [AI4Finance-Foundation/FinRL#928]
Hi everyone,

In traditional RL, every step consists of (S, A -> S', R), but in the context of FinRL, an action taken, i.e., buy or sell at whatever quantity, does NOT transfer one state to another, i.e., that an agent takes an action does not result in a price change from today to tomorrow.
If so, how should we calculate or estimate the price at every step?

Btw, is ElegantRL functioning perfectly just like SB3?

Please help confirm or correct me!

Thanx.

**Comment by XiaoYangLiu-FinRL** [comment]
1. Agent's state includes its own state and state of market. An action changes the "inner state" of the agent, NOT the state of a market.   
2. You can check the paper "reward is enough", where "reward maximizing is enough for learning intelligent behavior".  In essence, MDP is not a must for training deep neural network;  feeding data is a must.  Maybe we can forget about MDP, lol.

**Comment by julianzero** [comment]
> 1. Agent's state includes its own state and state of market. An action changes the "inner state" of the agent, NOT the state of a market.
> 2. You can check the paper "reward is enough", where "reward maximizing is enough for learning intelligent behavior".  In essence, MDP is not a must for training deep neural network;  feeding data is a must.  Maybe we can forget about MDP, lol.

The market states are the different prices plus technical indicators, but what are the "inner states" then?
Also, is ElegantRL working perfectly now? The actions output by SB3 do NOT look right, so I am betting on ElegantRL.

Thanx.
